src/dnaio/pairedend.py

```diff
@@ -47,26 +48,25 @@ def __iter__(self) -> Iterator[Tuple[Sequence, Sequence]]:
         """
         Iterate over the paired reads. Each item is a pair of Sequence objects.
         """
-        # Avoid usage of zip() below since it will consume one item too many.
+        # Avoid usage of zip() below since it will consume one item too many,
+        # when one of the iterators is exhausted. zip in python 3.10 has a
+        # 'strict' keyword that can be used to prevent this and throw an error,
+        # but it will take a long time for 3.10 or higher to be available on
+        # everyone's machine.
+        # Instead use zip_longest from itertools. This yields None if one of
+        # the iterators is exhausted. Checking for truthy values is speedy.
+        # So we can quickly check if the iterator is still yielding.
+        # This is faster than implementing a while loop with next calls,
+        # which requires expensive function lookups.
         it1, it2 = iter(self.reader1), iter(self.reader2)
-        while True:
-            try:
-                r1 = next(it1)
-            except StopIteration:
-                # End of file 1. Make sure that file 2 is also at end.
-                try:
-                    next(it2)
-                    raise FileFormatError(
-                        "Reads are improperly paired. There are more reads in "
-                        "file 2 than in file 1.",
-                        line=None,
-                    ) from None
-                except StopIteration:
-                    pass
-                break
-            try:
-                r2 = next(it2)
-            except StopIteration:
+        for r1, r2 in itertools.zip_longest(it1, it2):
+            if not r1:
```

Comment:
I think this needs to be `if r1 is None:`:
```python
>>> from dnaio import Sequence
>>> bool(Sequence("empty", "", ""))
False
```

Can you add a test?

Reply:
Ah, that would have gone horribly wrong indeed. It is now `is None` and a test was added.